Add TypeScript Guides (#6454)
DevExpress/devextreme-documentation: concepts/Common/TypeScript Guides/05 Scopes of Types/00 Scopes of Types.md

> When working with DevExtreme components in TypeScript, the correct use of types is essential for application creation. DevExtreme types may belong to one component, be common across several components, or be internal to the component suite.

Reviewer: I didn't really understand the following phrases: 

- "essential for application creation" (seems like such a generic phrase - almost anything you do affects application creation)
- "internal to the component suite" (I first thought that the scopes are: a single component, multiple components, component suite (even more components). But I later realized that the word "internal" was supposed to mean "do not use". At first, it seemed that "belongs to a component" and "internal to the suite" are similar expressions.)
 
```suggestion
If you build an application with DevExtreme components in TypeScript, you need to work with DevExpress types. Review the following sections to learn how to correctly use component-specific types and types shared by multiple controls/libraries. You will also learn how to identify internal DevExpress infrastructure types that you should not use in your applications. 
```

> If you build an application with DevExtreme components in TypeScript, you need to work with DevExpress types. Review the following sections to learn how to correctly use component-specific types and types shared by multiple controls/libraries. You will also learn how to identify internal DevExpress infrastructure types that you should not use in your applications. 

Reviewer 2: ```suggestion
If you build an application with DevExtreme components in TypeScript, you need to work with DevExpress types. Review the following sections for instructions on how to correctly use component-specific types and types shared by multiple controls/libraries. It also explains how to identify internal DevExpress infrastructure types that you should avoid in your applications. 
```
